spec/support/fake_analytics_spec.rb

```diff
@@ -0,0 +1,455 @@
+
+        expect { code_under_test.call }.
+          to raise_error(RSpec::Expectations::ExpectationNotMetError) do |err|
+            assert_error_messages_equal(err, <<~MESSAGE)
+              Expected that FakeAnalytics would have received event :my_event
+              with #<RSpec::Mocks::ArgumentMatchers::HashIncludingMatcher:<id> @expected={:arg1=>42}>.
+
+              Events received:
+              {:my_other_event=>[{}]}
+            MESSAGE
+          end
+      end
+
+      it 'raises if only a non-matching event of the same type has been logged' do
+        track_event_with_different_args.call
+
+        expect { code_under_test.call }.
+          to raise_error(RSpec::Expectations::ExpectationNotMetError) do |err|
+            assert_error_messages_equal(err, <<~MESSAGE)
+              Expected that FakeAnalytics would have received event :my_event
+              with #<RSpec::Mocks::ArgumentMatchers::HashIncludingMatcher:<id> @expected={:arg1=>42}>.
+
+              Events received:
+              {:my_event=>[{:arg1=>43}]}
+            MESSAGE
+          end
+      end
+
+      it 'does not raise if matching + non-matching event logged' do
+        track_event.call
+        track_event_with_different_args.call
+
+        expect { code_under_test.call }.
+          not_to raise_error(RSpec::Expectations::ExpectationNotMetError)
+      end
+
+      it 'does not raise if event was logged 1x' do
+        track_event.call
+        expect { code_under_test.call }.
+          not_to raise_error(RSpec::Expectations::ExpectationNotMetError)
+      end
+
+      it 'does not raise if event was logged 1x' do
+        track_event.call
+        expect { code_under_test.call }.
+          not_to raise_error(RSpec::Expectations::ExpectationNotMetError)
+      end
+
+      it 'does not raise if event was logged 2x' do
+        track_event.call
+        track_event.call
+
+        expect { code_under_test.call }.
+          not_to raise_error(RSpec::Expectations::ExpectationNotMetError)
+      end
+    end
+  end
+
+  describe FakeAnalytics::PiiAlerter do
+    it 'throws an error when pii is passed in' do
+      expect { analytics.track_event('Trackable Event') }.to_not raise_error
+
+      expect { analytics.track_event('Trackable Event', first_name: 'Bobby') }.
+        to raise_error(FakeAnalytics::PiiDetected)
+
+      expect do
+        analytics.track_event('Trackable Event', nested: [{ value: { first_name: 'Bobby' } }])
+      end.to raise_error(FakeAnalytics::PiiDetected)
+
+      expect { analytics.track_event('Trackable Event', decrypted_pii: '{"first_name":"Bobby"}') }.
+        to raise_error(FakeAnalytics::PiiDetected)
+    end
+
+    it 'throws an error when it detects sample PII in the payload' do
+      expect { analytics.track_event('Trackable Event', some_benign_key: 'FAKEY MCFAKERSON') }.
+        to raise_error(FakeAnalytics::PiiDetected)
+    end
+  end
+
+  describe FakeAnalytics::UndocumentedParamsChecker do
```

Comment:
thx for moving these over! I think we still need to delete them from `spec/services/analytics_specr.rb`

Reply:
Whoopsie